actors/core/src/main/java/com/ea/orbit/actors/ObserverManager.java

```diff
@@ -98,24 +102,35 @@ public boolean addObserver(final T observer)
      * It's not recommended to wait on this task since it might take a while to finish the
      * cleanup process and ObserverManager is thread safe.
      */
-    public Task<?> cleanup()
-    {
+    @SuppressWarnings("unchecked")
+    public Task<?> cleanup() {
         // TODO: replace ping for a single batch call for each node containing observers from this list.
         // TODO: add a observer validation function to the runtime. ActorRuntime
-        final Stream<Task<?>> stream = observers.stream()
-                .map(o ->
-                        ((Task<?>) (o).ping()).whenComplete((final Object pr, final Throwable pe) ->
-                        {
-                            if (pe != null)
+        Stream stream = this.observers.stream()
+                .map(observer ->
+                        (observer.ping()).handle((Object result, Throwable ex) -> {
+                            if (ex != null)
                             {
-                                // beware: this might run in parallel with other calls the actor.
-                                // this shouldn't be a problem.
-                                observers.remove(o);
-                            }
-                        }));
+                                return checkAlive(observer).thenAccept(alive -> {
```

Comment:
We should remove the observers when they no longer exist in the remote node.
This change will let old zombie observers exist until their original node restarts.

I'll take this change to another branch and add the method @johnou cited:
a call to the original node confirming that the observer is alive.
this will remove the ping.

The new flow will be:
cleanup:
- is the original node still up? no: remove observer
- is the observer alive in the node? no: remove observer.
- keep observer